screens/role/ConfigPrivilege.tsx

```diff
@@ -0,0 +1,131 @@
+import { CompositeScreenProps } from "@react-navigation/native";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { Controller, FieldValues, useForm } from "react-hook-form";
+import { Alert, StyleSheet, View } from "react-native";
+
+import ChangeScreenTab from "@/components/button/ChangeScreenTab";
+import PrimaryButton from "@/components/button/PrimaryButton";
+import SecondaryButton from "@/components/button/SecondaryButton";
+import RoleCard from "@/components/card/RoleCard";
+import TextInput from "@/components/input/TextInput";
+import SubHeaderText from "@/components/text/SubHeaderText";
+import BodyContainer from "@/components/ui/BodyContainer";
+import { ActionMode } from "@/enum/ActionMode";
+import { ManagingCategory } from "@/enum/ManagingCategory";
+import { OtherStackParamList, AuthenticatedStackParamList } from "@/types";
+
+export type ConfigPrivilegeProps = CompositeScreenProps<
+  NativeStackScreenProps<OtherStackParamList, "ConfigPrivilege">,
+  NativeStackScreenProps<AuthenticatedStackParamList>
+>;
+
+const ConfigPrivilege: React.FC<ConfigPrivilegeProps> = ({ navigation }) => {
+  const category = ManagingCategory.PRIVILEGE;
+  const form = useForm();
+  const { control, handleSubmit } = form;
+
+  const onSubmit = async (data: FieldValues) => {
+    try {
+      // todo: send data
+      console.log("data", data);
+    } catch (error) {
+      Alert.alert(
+        "Create request error",
+        "Please try again!!: " + (error as Error).message
+      );
+    }
+  };
+
+  const mock = [{}];
+
+  return (
+    <BodyContainer innerContainerStyle={styles.container}>
+      <Controller
+        name={"name"}
+        control={control}
+        render={({ field: { onChange, value } }) => (
+          <TextInput
+            title="Name"
+            placeholder="Enter your privilege name"
+            value={value}
+            onChangeText={(value) => onChange(value)}
+          />
+        )}
+      />
+      <Controller
+        name={"description"}
+        control={control}
+        render={({ field: { onChange, value } }) => (
+          <TextInput
+            title="Description"
+            placeholder="Enter your privilege description"
+            value={value}
+            onChangeText={(value) => onChange(value)}
+          />
+        )}
+      />
+      <SubHeaderText text="Privilege" />
+      <View style={styles.roleCardContainer}>
+        {mock.map((_, index) => (
+          <RoleCard
+            category={category}
+            roleName="A1"
+            description="60 baht/hr"
+            key={index}
+            onPress={() =>
+              navigation.navigate("ConfigZone", {
+                form: form,
+                mode: ActionMode.EDIT,
+                index: index,
+              })
+            }
+          />
+        ))}
```

Comment:
Can you make it more clear when we connect to backend? Please do not think that we use mock here but you have to think how we transfer the data from which place to which place

Reply:
I saw many places like this. Please fix all of them